### PR TITLE
Use discussion subdomain

### PR DESCRIPTION
**Pull request by nicl**
The result of this change is that we no longer need to proxy through the
www.theguardian.com subdomain; all the special logic here can be
removed once this goes live.

As part of this PR two switches are removed as they are no longer
required:
- discussion https flag (we are https-only now as our auth mechanism
  requires this)
- discussion proxy (the new subdomain and also move to always-https mean
  that the proxy is no longer required)

cc @guardian/discussion @johnduffell

### CODE REVIEW

File: discussion/test/package.scala [hunk 1]
```diff
@@ -57,7 +58,13 @@ object DiscussionApiHttpRecorder extends HttpRecorder[WSResponse] {
...
   import play.api.Play.current
   protected val clientHeaderValue: String =""
-  protected val apiRoot = conf.Configuration.discussion.apiRoot
+
+  protected val apiRoot =
+    if (Configuration.environment.isProd)
+      Configuration.discussion.apiRoot
+    else
+      Configuration.discussion.apiRoot.replaceFirst("https://", "http://") // CODE SSL cert is defective and expensive to fix
+
```
Comment (jennysivapalan) [inline discussion/test/package.scala]: If this is a test don't we just want the non prod one?

**Comment by jennysivapalan** [comment]
:+1: :speech_balloon:

**Comment by satterly** [comment]
:+1:

File: discussion/test/package.scala (hunk 1 above)
Comment (nicl) [inline discussion/test/package.scala]: Hmm, maybe. I wasn't sure what people might do for testing!

**Pull request merged**